lib/instrumentation/index.ts

```diff
@@ -0,0 +1,91 @@
+import { ClientRequest } from 'http';
+import type { Tracer } from '@opentelemetry/api';
+import * as api from '@opentelemetry/api';
+import { SpanStatusCode } from '@opentelemetry/api';
+import { AsyncLocalStorageContextManager } from '@opentelemetry/context-async-hooks';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import { BunyanInstrumentation } from '@opentelemetry/instrumentation-bunyan';
+import { HttpInstrumentation } from '@opentelemetry/instrumentation-http';
+import { Resource } from '@opentelemetry/resources';
+import {
+  BatchSpanProcessor,
+  ConsoleSpanExporter,
+  SimpleSpanProcessor,
+} from '@opentelemetry/sdk-trace-base';
+import { NodeTracerProvider } from '@opentelemetry/sdk-trace-node';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { pkg } from '../expose.cjs';
+import {
+  isTraceDebuggingEnabled,
+  isTraceSendingEnabled,
+  isTracingEnabled,
+} from './utils';
+
+let traceProvider: NodeTracerProvider;
+
+init();
+
+export function init(): void {
+  traceProvider = new NodeTracerProvider({
+    resource: new Resource({
+      // https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/resource/semantic_conventions/README.md#semantic-attributes-with-sdk-provided-default-value
+      [SemanticResourceAttributes.SERVICE_NAME]: 'renovate',
+      [SemanticResourceAttributes.SERVICE_NAMESPACE]: 'renovate.com',
+      [SemanticResourceAttributes.SERVICE_VERSION]: pkg.version,
+    }),
+  });
+  api.trace.setGlobalTracerProvider(traceProvider);
+
+  // add processors
+  if (isTraceDebuggingEnabled()) {
+    traceProvider.addSpanProcessor(
+      new SimpleSpanProcessor(new ConsoleSpanExporter())
```

Comment:
Is this using `console` to log, or does it inject itself into bunyan?

Reply:
It writes to console as this code has to run before bunyan is initialized.